google/cloud/apigeeregistry/v1/admin_service.proto

```diff
@@ -105,6 +117,23 @@ message Status {
   string message = 1;
 }
 
+// Request message for MigrateDatabase.
+message MigrateDatabaseRequest {
+  // A string describing the kind of migration to perform.
+  // Currently only "auto" is recognized (and is the default if omitted).
+  string kind = 1;
```

Comment:
Do we expect this to change frequently? [AIP-126](https://google.aip.dev/126) recommends enums unless we'll be frequently modifying the possible values here. 

As a caveat, I think we should use enum unless we think it will change frequently _and_ have clients using older versions of this proto to call newer versions of the admin code. We don't really have to worry about breaking changes to enum fields until there's somebody to break with the change.

Reply:
A string value allows us to add options without changing the API - as you noted, adding enums are breaking changes and would require regenerating/rebuilding API clients. 

I'm not sure what we might agree is "frequent", but I think adding a custom database migration might be a change that we want to make very quickly (perhaps responding to a problem), so the less overhead associated with that, the better.

Reply:
From on [AIP-180](https://google.aip.dev/180): Enum values may be freely added to enums which are only used in request messages. So I'd like to understand the `breaking change` part better.

Let's say we're using enum for `kind`, and we're adding a new enum value besides `auto` (e.g. `partial`); under this case, if a user calls the new version of the admin code from an old API client, would the user still be able to make successful MigrateDatabase request with `kind = auto`?

Reply:
@wangyh28 Yes, but the user wouldn't be able to pass the new `kind` value without regenerating/rebuilding their code. If we use a string field for `kind`, then it is just a configuration change (assuming that the code that calls the migration API is written to read a configured value for `kind`)

Reply:
@seaneganx If you've gotten a copy of [API Design Patterns](https://www.manning.com/books/api-design-patterns), you can read JJ's discussion of enums there, which is more strongly cautionary than the warnings in AIP-126. From the conclusion of Chapter 5: "Enumerations should generally be avoided, relying on string values instead, with validation done on the server side rather than the client side." That''s expanded earlier in section 5.5, "while [enumerations] might be wonderful things in languages like Java, their transplantation into the world of web APIs is often a mistake."